doc/source/whatsnew/v0.17.1.txt

```diff
@@ -26,7 +26,7 @@ Enhancements
 
 Other Enhancements
 ^^^^^^^^^^^^^^^^^^
-
+- raising ``NotImplementedError`` for ``Index.shift()`` when no ``DatetimeIndex`` is used.
```

Comment:
say: raise ``NotImplementedError`` in ``Index.shift`` for non-supported index types. (and include the issue number)